### PR TITLE
Add source-kafka plugin

### PR DESCRIPTION
**Pull request by dsimansk**
Verify source-kafka plugin works with S-O based Kafka instance.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -16,9 +17,10 @@ require (
 	k8s.io/cli-runtime v0.18.8
 	k8s.io/client-go v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible
 	k8s.io/code-generator v0.18.8
-	knative.dev/eventing v0.18.0
+	knative.dev/eventing v0.18.5-0.20201105155307-650096a39064
+	knative.dev/kn-plugin-source-kafka v0.0.0-20201203183209-e1f755efaca3
```
Comment (navidshaikh) [inline go.mod:21]: we'll get the release done tomorrow to have this pinned at a tag

File: go.mod [hunk 2]
```diff
@@ -16,9 +17,10 @@ require (
...
 	knative.dev/networking v0.0.0-20200922180040-a71b40c69b15
-	knative.dev/pkg v0.0.0-20200922164940-4bf40ad82aab
+	knative.dev/pkg v0.0.0-20201026165741-2f75016c1368
```
Comment (navidshaikh) [inline go.mod:23]: hmm, thats coming from plugin

File: openshift/e2e-common.sh [hunk 3]
```diff
@@ -204,6 +204,24 @@ install_serverless_operator_branch() {
   unset OPENSHIFT_CI
   ./hack/install.sh || failed=1
   subheader "Successfully installed serverless operator."
+
+  subheader "Install Kafka from serverless-operator repo"
+  INSTALL_SERVING="false" INSTALL_KAFKA="true" ./hack/install.sh || failed=1
+  subheader "Successfully installed serverless operator."
```
Comment (navidshaikh) [inline openshift/e2e-common.sh:210]:
```suggestion
  subheader "Successfully installed kafka."
```

File: test/e2e/source_kafka_test.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
+// Copyright 2020 The Knative Authors
```
Comment (navidshaikh) [inline test/e2e/source_kafka_test.go:1]: :rocket:

**Review by navidshaikh — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (dsimansk) [reply go.mod:21]: Yep, it's pulling `master` version directly. I'm expecting some nits in Kafka setup, therefore though might be a good idea to give a head start.

File: go.mod (hunk 2 above)
Comment (dsimansk) [reply go.mod:23]: Indeed, it's pulled through eventing-contrib upgrade to `0.18.6`.

File: openshift/e2e-common.sh [hunk 5]
```diff
@@ -204,6 +204,24 @@ install_serverless_operator_branch() {
   unset OPENSHIFT_CI
   ./hack/install.sh || failed=1
   subheader "Successfully installed serverless operator."
+
+  subheader "Install Kafka from serverless-operator repo"
+  INSTALL_SERVING="false" INSTALL_KAFKA="true" ./hack/install.sh || failed=1
+  subheader "Successfully installed kafka."
+
+  header "Applying Strimzi Topic CR"
+  cat <<-EOF | oc apply -f - || failed=1
```
Comment (navidshaikh) [inline openshift/e2e-common.sh:213]:
```suggestion
  cat <<-EOF | oc apply -n kafka -f - || failed=1
```
@dsimansk : I think we'd need to mention the namespace here.

Comment (dsimansk) [reply openshift/e2e-common.sh]: Ahh, that's a great point. I'm about to test it on CRC, I'll try to add namespace as well.

**Comment by dsimansk** [comment]
The last missing piece is a tagged release from https://github.com/knative-sandbox/kn-plugin-source-kafka/releases. When available I'll update the PR and remove WIP as well.

**Comment by navidshaikh** [comment]
/retest

level=error msg="Cluster operator authentication Degraded is True with IngressStateEndpoints_MissingEndpoints::RouteStatus_FailedHost: IngressStateEndpointsDegraded: No endpoints found for oauth-server\nRouteStatusDegraded: route is not available at canonical host oauth-openshift.apps.ci-op-fvs2v1p6-4cc3a.origin-ci-int-aws.dev.rhcloud.com: []"

**Review by navidshaikh — APPROVED** [review]
/lgtm

**Pull request merged**